Formula/h/haskell-stack.rb

```diff
@@ -22,30 +22,33 @@ class HaskellStack < Formula
   end
 
   depends_on "cabal-install" => :build
-  depends_on "ghc@9.2" => :build
+  depends_on "ghc" => :build
 
   uses_from_macos "zlib"
 
-  # All ghc versions before 9.2.1 requires LLVM Code Generator as a backend on
-  # ARM. GHC 8.10.7 user manual recommend use LLVM 9 through 12 and we met some
-  # unknown issue with LLVM 13 before so conservatively use LLVM 12 here.
-  #
-  # References:
-  #   https://downloads.haskell.org/~ghc/8.10.7/docs/html/users_guide/8.10.7-notes.html
-  #   https://gitlab.haskell.org/ghc/ghc/-/issues/20559
-  on_arm do
-    depends_on "llvm@12"
-  end
-
   def install
     # Remove locked dependencies which only work with a single patch version of GHC.
     # If there are issues resolving dependencies, then can consider bootstrapping with stack instead.
     (buildpath/"cabal.project").unlink
+    (buildpath/"cabal.project").write <<~EOS
+      packages: .
+    EOS
 
     system "cabal", "v2-update"
     system "cabal", "v2-install", *std_cabal_v2_args
+  end
 
-    bin.env_script_all_files libexec, PATH: "${PATH}:#{Formula["llvm@12"].opt_bin}" if Hardware::CPU.arm?
+  def caveats
+    on_macos do
+      on_arm do
+        <<~EOS
+          All GHC versions before 9.2.1 requires LLVM Code Generator as a backend
+          on ARM. If you are using one of those GHC versions with `haskell-stack`,
+          then you may need to install a supported LLVM version and add its bin
+          directory to the PATH.
+        EOS
```

Comment:
Not too sure if best way to handle but we did remove `llvm(@*)` dependency in `ghc` 9.8, 9.6, and 9.4. So, rather than forcing old `llvm@12` dependency, I was thinking it may be better to just leave as caveat for users who want less/un-supported versions of GHC.

Reply:
LGTM